enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -90,16 +90,48 @@ const (
 	// flushing any atime updates in an incomplete batch (that have not
 	// already been flushed due to throughput)
 	atimeFlushPeriod = 10 * time.Second
+
+	// Default values for options
+	defaultBlockCacheSizeBytes    = int64(1000 * megabyte)
+	defaultMaxInlineFileSizeBytes = int64(1024)
+	defaultAtimeUpdateThreshold   = 10 * time.Minute
+	defaultAtimeWriteBatchSize    = 1000
+	defaultAtimeBufferSize        = 100000
+	defaultMinEvictionAge         = 6 * time.Hour
 )
 
-// Options is a struct containing the pebble cache configuration options.
+// options is a struct containing the pebble cache configuration options.
 // Once a cache is created, the options may not be changed.
-type Options struct {
-	RootDirectory       string
-	Partitions          []disk.Partition
-	PartitionMappings   []disk.PartitionMapping
-	MaxSizeBytes        int64
-	BlockCacheSizeBytes int64
+type options struct {
+	RootDirectory     string
+	Partitions        []disk.Partition
+	PartitionMappings []disk.PartitionMapping
+
+	MaxSizeBytes           int64
+	BlockCacheSizeBytes    int64
+	MaxInlineFileSizeBytes int64
+
+	AtimeUpdateThreshold time.Duration
+	AtimeWriteBatchSize  int
+	AtimeBufferSize      int
+	MinEvictionAge       time.Duration
+}
+
+// OptionsConstructor contains fields required to initialize an options struct
+// If a field is nil, the corresponding field in the options struct will be set to the default value
+type OptionsConstructor struct {
+	RootDirectory     *string
```

Comment:
Convention in Go is to use empty values (i.e. "" for string, 0 for integers) to indicate whether something is set. It's less common to use pointers.

Reply:
The problem I was facing is that sometimes we explicitly want to set some of these values to 0. So I was having trouble detecting whether something was unset and should be set to the default, or was explicitly set to 0 and so we shouldn't overwrite it with the default value. Is there a more idiomatic way to handle that?

Reply:
You can use pointers for the cases where 0 is a valid value, but I would recommend using plain values in all other cases.

Reply:
Updated. Do you find it confusing that certain fields are pointers but others are just the plain values?